tests/integration/generate_fixtures/common.py

```diff
@@ -0,0 +1,240 @@
+
+GENESIS_DATA = {
+    "config": {
+        "chainId": 1337,
+        "homesteadBlock": 0,
+        "eip150Block": 0,
+        "eip155Block": 10,
+        "eip158Block": 10,
+        "eip160Block": 10
+    },
+    "nonce": "0x0000000000000042",
+    "alloc": {
+        COINBASE: {
+            "balance": "1000000000000000000000000000"
+        },
+        UNLOCKABLE_ACCOUNT: {
+            "balance": "1000000000000000000000000000"
+        },
+        RAW_TXN_ACCOUNT: {
+            "balance": "1000000000000000000000000000"
+        }
+    },
+    "timestamp": "0x00",
+    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "extraData": "0x3535353535353535353535353535353535353535353535353535353535353535",
+    "gasLimit": "0x1000000",
+    "difficulty": "0x10000",
+    "mixhash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "coinbase": COINBASE
+}
+
+
+def ensure_path_exists(dir_path):
+    """
+    Make sure that a path exists
+    """
+    if not os.path.exists(dir_path):
+        os.makedirs(dir_path)
+        return True
+    return False
+
+
+@contextlib.contextmanager
+def tempdir():
+    dir_path = tempfile.mkdtemp()
+    try:
+        yield dir_path
+    finally:
+        shutil.rmtree(dir_path)
+
+
+def get_open_port():
+    sock = socket.socket()
+    sock.bind(('127.0.0.1', 0))
+    port = sock.getsockname()[1]
+    sock.close()
+    return str(port)
+
+
+def get_geth_binary():
+    from geth.install import (
+        get_executable_path,
+        install_geth,
+    )
+
+    if 'GETH_BINARY' in os.environ:
+        return os.environ['GETH_BINARY']
+    elif 'GETH_VERSION' in os.environ:
+        geth_version = os.environ['GETH_VERSION']
+        _geth_binary = get_executable_path(geth_version)
+        if not os.path.exists(_geth_binary):
+            install_geth(geth_version)
+        assert os.path.exists(_geth_binary)
+        return _geth_binary
+    else:
+        return 'geth'
+
+
+def wait_for_popen(proc, timeout):
+    start = time.time()
+    while time.time() < start + timeout:
+        if proc.poll() is None:
+            time.sleep(0.01)
+        else:
+            break
+
+
+def kill_proc_gracefully(proc):
+    if proc.poll() is None:
+        proc.send_signal(signal.SIGINT)
+        wait_for_popen(proc, 13)
+
+    if proc.poll() is None:
+        proc.terminate()
+        wait_for_popen(proc, 5)
+
+    if proc.poll() is None:
+        proc.kill()
+        wait_for_popen(proc, 2)
+
+
+def wait_for_socket(ipc_path, timeout=30):
+    start = time.time()
+    while time.time() < start + timeout:
+        try:
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            sock.connect(ipc_path)
+            sock.settimeout(timeout)
+        except (FileNotFoundError, socket.error):
+            time.sleep(0.01)
+        else:
+            break
+
+
+@contextlib.contextmanager
+def get_geth_process(geth_binary,
```

Comment:
Same her, `go_ethereum.py` instead of `common.py`?

Reply:
I put those in common.py because I use them in the parity script to create the cluster.

Reply:
Took me a minute to figure out what was going on.  That makes sense now.  +1 for clever way to de-duplicate the logic for initializing the chain.